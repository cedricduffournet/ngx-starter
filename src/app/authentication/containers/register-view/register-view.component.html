<app-auth-layout title="{{ 'REGISTER' | translate }}">
  <ng-container *ngIf="status$ | async as status">
    <app-register-form
      *ngIf="status === 'init' || (errorUserExist$ | async)"
      [processing]="processing$ | async"
      [civilities]="civilities$ | async"
      [errorUserExist]="errorUserExist$ | async"
      (register)="onRegister($event)"
    ></app-register-form>
    <div
      *ngIf="status === 'success'"
      class="alert alert-success"
      innerHTML="{{
        'REGISTER_SUCCESS' | translate: { email: getEmailRegistered() }
      }}"
    ></div>
    <a [routerLink]="['/login']" class="btn btn-link btn-block" translate>
      CONNECT
    </a>
  </ng-container>
</app-auth-layout>
