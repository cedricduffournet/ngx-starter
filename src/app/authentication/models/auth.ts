export interface AuthToken {
  access_token: string;
  expires_in: number;
  token_type: string;
  refresh_token: string;
}

export interface Credentials {
  login: string;
  password: string;
}

export interface ResetPassword {
  plainPassword: {
    first: string;
    second: string;
  };
}

export interface Register extends ResetPassword {
  civility: number;
  firstname: string;
  lastname: string;
  email: string;
  confirmationUrl: string;
}

export interface SendResetPassword {
  username: string;
}
