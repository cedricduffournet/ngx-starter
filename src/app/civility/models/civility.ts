import { schema } from 'normalizr';

export interface Civility {
  id: number;
  name: string;
  code: string;
}

export const civilitySchema = new schema.Entity('civilities');
