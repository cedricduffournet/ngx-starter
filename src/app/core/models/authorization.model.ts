export interface Authorization {
  create: boolean;
  update: boolean;
  delete: boolean;
}
