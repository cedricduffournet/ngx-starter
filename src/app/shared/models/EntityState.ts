export interface EntityState<T> {
  [key: string]: T;
}
