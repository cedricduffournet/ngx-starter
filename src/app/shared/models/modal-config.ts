import { ModalOptions } from 'ngx-bootstrap/modal/modal-options.class';

export const GLOBAL_MODAL_CONFIG: ModalOptions = {
  animated: false
};

export const CRUD_MODAL_CONFIG: ModalOptions = {
  ...GLOBAL_MODAL_CONFIG,
  keyboard: false,
  ignoreBackdropClick: true
};
