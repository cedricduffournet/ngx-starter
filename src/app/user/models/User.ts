import { Civility } from '@app/civility/models/civility';

export interface UserGroup {
  name: string;
  roles: string[];
  superAdmin: boolean;
  admin: boolean;
}

export interface User {
  id: number;
  firstname: string;
  lastname: string;
  email: string;
  civility: Civility;
  roles: string[];
  groups: UserGroup[];
  enabled: boolean;
}
