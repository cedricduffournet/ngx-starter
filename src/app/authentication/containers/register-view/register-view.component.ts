import {
  ChangeDetectionStrategy,
  Component,
  OnInit,
  OnDestroy
} from '@angular/core';

import { Observable } from 'rxjs';
import { Store, select } from '@ngrx/store';

import { User } from '@app/user/models/User';
import { Register } from '@app/authentication/models/auth';
import { RegisterViewActions } from '@app/authentication/state/actions';
import * as fromAuth from '@app/authentication/state/reducers';
import * as fromCivilities from '@app/civility/state/reducers';
import { Civility } from '@app/civility/models/civility';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'app-register-view',
  templateUrl: 'register-view.component.html'
})
export class RegisterViewComponent implements OnInit, OnDestroy {
  registered: Register;
  processing$: Observable<boolean>;
  status$: Observable<'init' | 'success' | 'error'>;
  errorUserExist$: Observable<boolean>;
  userRegistered$: Observable<User | null>;
  civilities$: Observable<Civility[]>;
  username: string;
  errorMail = false;
  msgErrorMail = '';

  constructor(public store: Store<fromAuth.State & fromCivilities.State>) {}

  ngOnInit() {
    this.processing$ = this.store.pipe(
      select(fromAuth.getRegisterViewProcessing)
    );
    this.status$ = this.store.pipe(select(fromAuth.getRegisterViewStatus));
    this.errorUserExist$ = this.store.pipe(
      select(fromAuth.getRegisterViewErrorUserExist)
    );
    this.userRegistered$ = this.store.pipe(
      select(fromAuth.getRegisterViewUserRegistered)
    );
    this.civilities$ = this.store.pipe(select(fromCivilities.getCivilities));
    this.store.dispatch(RegisterViewActions.loadCivilities());
  }

  ngOnDestroy() {
    this.store.dispatch(RegisterViewActions.clear());
  }

  onRegister(register: Register) {
    this.registered = register;
    this.store.dispatch(RegisterViewActions.register({ register }));
  }

  getEmailRegistered() {
    return this.registered.email;
  }
}
