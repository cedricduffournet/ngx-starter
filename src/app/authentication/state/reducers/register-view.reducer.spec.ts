import { reducer } from '@app/authentication/state/reducers/register-view.reducer';
import * as fromRegisterView from '@app/authentication/state/reducers/register-view.reducer';
import { User } from '@app/user/models/User';
import { Register } from '@app/authentication/models/auth';
import {
  RegisterViewActions,
  AuthApiActions
} from '@app/authentication/state/actions';

describe('RegisterViewReducer ', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const action = {} as any;

      const result = reducer(undefined, action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('CLEAR', () => {
    it('shoud restore state to initial state', () => {
      const initialState = {
        processing: true,
        errorUserExist: true,
        status: 'success',
        userRegistered: {
          firstname: 'firstname'
        }
      } as fromRegisterView.State;

      const action = RegisterViewActions.clear();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('REGISTER', () => {
    it('should set processing to true', () => {
      const initialState = {
        processing: true
      } as fromRegisterView.State;

      const register = {
        firstname: 'firstname'
      } as Register;

      const action = RegisterViewActions.register({ register });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('REGISTER_SUCCESS', () => {
    it('should set processing to false, status to success and userRegistered', () => {
      const initialState = {
        processing: true,
        userRegistered: null
      } as fromRegisterView.State;

      const userRegistered = {
        firstname: 'firstname'
      } as User;

      const action = AuthApiActions.registerSuccess({ userRegistered });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('REGISTER_FAILURE', () => {
    it('should set processing to false,status to error and userExist to true', () => {
      const initialState = {
        processing: true,
        userRegistered: null,
        status: 'init'
      } as fromRegisterView.State;

      const action = AuthApiActions.registerFailure({ error: true });
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });
});
