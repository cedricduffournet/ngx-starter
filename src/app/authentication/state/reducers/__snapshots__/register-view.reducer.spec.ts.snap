// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterViewReducer  CLEAR shoud restore state to initial state 1`] = `
Object {
  "errorUserExist": false,
  "processing": false,
  "status": "init",
  "userRegistered": null,
}
`;

exports[`RegisterViewReducer  REGISTER should set processing to true 1`] = `
Object {
  "errorUserExist": false,
  "processing": true,
  "status": "init",
  "userRegistered": null,
}
`;

exports[`RegisterViewReducer  REGISTER_FAILURE should set processing to false,status to error and userExist to true 1`] = `
Object {
  "errorUserExist": true,
  "processing": false,
  "status": "error",
  "userRegistered": null,
}
`;

exports[`RegisterViewReducer  REGISTER_SUCCESS should set processing to false, status to success and userRegistered 1`] = `
Object {
  "errorUserExist": false,
  "processing": false,
  "status": "success",
  "userRegistered": Object {
    "firstname": "firstname",
  },
}
`;

exports[`RegisterViewReducer  undefined action should return the default state 1`] = `
Object {
  "errorUserExist": false,
  "processing": false,
  "status": "init",
  "userRegistered": null,
}
`;
